### ISSUE TITLE
System Shutdown: Out of Memory Failure

### ISSUE THREAD

**Issue by joker1138** [hoobs-org/HOOBS#517]
**Description**
System will shut down
**Version**
Please include the version of HOOBS you are using.
3.1.22
**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [X] Yes
- [ ] No

**Pervious version**
If you upgraded, please let us know your previous version.
Whatever was directly before 3.1.22

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [X] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [X] HOOBS SD Card
- [ ] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.

**List your plugins**
Cahmberlian my Q
TP Smart Link
Simplisafe

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
```
[Homebridge] <--- Last few GCs --->
[Homebridge] [708:0x25fa100] 111944689 ms: Mark-sweep 625.4 (740.1) -> 616.0 (740.1) MB, 1492.5 / 11.7 ms  (average mu = 0.282, current mu = 0.396) allocation failure scavenge might not succeed
[Homebridge] [708:0x25fa100] 111950130 ms: Mark-sweep 621.6 (740.1) -> 617.3 (736.1) MB, 4985.9 / 3.8 ms  (average mu = 0.156, current mu = 0.083) allocation failure scavenge might not succeed
[Homebridge] <--- JS stacktrace --->
[Homebridge] ==== JS stack trace =========================================
[Homebridge] 0: ExitFrame [pc: 0x491eae9c]
[Homebridge] 1: StubFrame [pc: 0x49197b40]
[Homebridge] Security context: 0x29adb7c1 <JSObject>
[Homebridge] 2: /* anonymous */ [0x43fc8099] [/home/hoobs/.hoobs/node_modules/socket.io-client/lib/socket.js:~136] [pc=0x9da54160](this=0x2f743c3d <Socket map = 0x24e086f5>,ev=0x43ffc74d <String[12]: reconnecting>)
[Homebridge] 3: arguments adaptor frame: 2->1
[Homebridge] 4: /* anonymous */ [0x5ad29c6d] [/home/hoobs/.hoobs/node_modules/socket.io-client/lib/m...
[Homebridge] FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
```

**Additional context**

**Comment by mkellsy** [comment]
I traced this in out code, and it looks like there is an issue with one of your plugins.

* Cahmberlian my Q
* TP Smart Link
* Simplisafe

I need your config to try to reproduce. Can you post your config. Remove all usernames and passwords.

**Comment by BobbySlope** [comment]
@joker1138 please post your config, otherwise we can't solve the issue

**Comment by mkellsy** [comment]
I just released 3.1.27
This issue has been fixed.
I recommend updating.
